libraries/Microsoft.Bot.StreamingExtensions/ConcurrentStream.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.StreamingExtensions.Payloads;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    internal class ConcurrentStream : Stream
+    {
```

Comment:
BUG: This class is riddled with concurrency bugs. It needs to be totally revisited.

Reply:
Not true. Do you have a case?

We have many concurrent tests evaluating this class, and it has been significantly reviewed for it's use cases in concurrent cases.

Reply:
We had a talk and decided to rename the class and add comments.